tools/hubl/internal/registry.go

```diff
@@ -0,0 +1,104 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+)
+
+type ChainRegistryEntry struct {
+	APIs struct {
+		GRPC []*APIEntry `json:"grpc"`
+	} `json:"apis"`
+}
+
+type APIEntry struct {
+	Address  string
+	Provider string
+}
+
+func GetChainRegistryEntry(chain string) (*ChainRegistryEntry, error) {
+	res, err := http.Get(fmt.Sprintf("https://raw.githubusercontent.com/cosmos/chain-registry/master/%v/chain.json", chain))
+	if err != nil {
+		return nil, err
+	}
+	bz, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = res.Body.Close(); err != nil {
+		return nil, err
+	}
+
+	data := &ChainRegistryEntry{}
+	if err = json.Unmarshal(bz, data); err != nil {
+		return nil, err
+	}
+
+	// clean-up the URL
+	cleanEntries := make([]*APIEntry, 0)
```

Comment:
nit: capacity could already be allocated
```suggestion
	cleanEntries := make([]*APIEntry, 0, len(data.APIs.GRPC)
```